File: backend/app/schemas/message.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime


class MessageCreateRequest(BaseModel):
    content: str
    content_type: str = "text"
    attachment_url: str | None = None


class MessageResponse(BaseModel):
    id: UUID
    chat_id: UUID
    role: str
    content: str
    content_type: str
    attachment_url: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class WSMessage(BaseModel):
    type: str  # "message", "typing", "stop_typing"
    chat_id: UUID | None = None
    content: str | None = None
    content_type: str = "text"
    attachment_url: str | None = None

File: app/lib/models/message.dart
class Message {
  final String id;
  final String chatId;
  final String role;
  String content;
  final String contentType;
  final String? attachmentUrl;
  final DateTime createdAt;
  bool isStreaming;

  Message({
    required this.id,
    required this.chatId,
    required this.role,
    required this.content,
    this.contentType = 'text',
    this.attachmentUrl,
    required this.createdAt,
    this.isStreaming = false,
  });

  factory Message.fromJson(Map<String, dynamic> json) => Message(
        id: json['id'] ?? json['message_id'] ?? '',
        chatId: json['chat_id'],
        role: json['role'],
        content: json['content'] ?? '',
        contentType: json['content_type'] ?? 'text',
        attachmentUrl: json['attachment_url'],
        createdAt: json['created_at'] != null
            ? DateTime.parse(json['created_at'])
            : DateTime.now(),
      );
}
